espnet/nets/pytorch_backend/transformer/attention.py

```diff
@@ -142,24 +144,32 @@ def rel_shift(self, x, zero_triu=False):
         """Compute relative positinal encoding.
 
         Args:
-            x (torch.Tensor): Input tensor (batch, time, size).
+            x (torch.Tensor): Input tensor (batch, head, time1, time2).
```

Comment:
`time1` and `time2` are confusing. Please clarify which is query/key lengths.

Reply:
Many thanks for the suggestions. I think `time1=query_len` and `time2=pos_emb_len`. Actually, the length of pos_emb is the same as query (use absolute distance) or `2*time1 - 1` (use positive and negative distance). I will clarify it.

Reply:
Hi, I'm confused why pos_emb len is the same as the query? What if query is short than key as illustrated in https://arxiv.org/pdf/1901.02860.pdf?